de.tu_bs.cs.isf.e4cf.text_editor/src/de/tu_bs/cs/isf/e4cf/text_editor/FileUtils.java

```diff
@@ -0,0 +1,211 @@
+package de.tu_bs.cs.isf.e4cf.text_editor;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import javax.inject.Inject;
+
+import de.tu_bs.cs.isf.e4cf.core.util.RCPContentProvider;
+import de.tu_bs.cs.isf.e4cf.core.util.RCPMessageProvider;
+import de.tu_bs.cs.isf.e4cf.core.util.ServiceContainer;
+import de.tu_bs.cs.isf.e4cf.text_editor.stringtable.EditorST;
+import javafx.stage.FileChooser;
+import javafx.stage.FileChooser.ExtensionFilter;
+import javafx.stage.Window;
+
+/**
+ * Contains methods for common file interactions such as opening and closing
+ * 
+ * @author Erwin Wijaya, Lukas Cronauer, Soeren Christmann, Cedric Kapalla
+ *
+ */
+public class FileUtils {
+	private FileChooser fileChooser;
+	private Window parent;
+	private int lastSavedRevision;
+
+	/**
+	 * Constructor used to initialize the fileChooser instance of this object.
+	 * Available file extensions are added to the fileChooser.
+	 * 
+	 * @param parent Window the @TextEditor is part of. This will be needed to
+	 *               display open/save dialogs
+	 * 
+	 * @author Lukas Cronauer, Erwin Wijaya
+	 */
+	public FileUtils(Window parent) {
+		this.parent = parent;
+
+		fileChooser = new FileChooser();
+
+		for (String extension : EditorST.FILE_FORMATS) {
+			String displayName = extension.substring(0, 1).toUpperCase() + extension.substring(1) + " File (."
+					+ extension + ")";
+			String extensionFormat = "*." + extension;
+			fileChooser.getExtensionFilters().add(new ExtensionFilter(displayName, extensionFormat));
+		}
+
+		fileChooser.setInitialDirectory(new File(RCPContentProvider.getCurrentWorkspacePath()));
+	}
+
+	/**
+	 * Extracts the fileName from a string containing a file path.
+	 * 
+	 * @param path A filePath ending in a file
+	 * @return The fileName with extension (e.g. name.extension)
+	 * @author Lukas Cronauer
+	 */
+	public static String parseFileNameFromPath(String path) {
+		String fileName = path;
+		String[] splittedPath;
+		if (!path.startsWith(EditorST.NEW_TAB_TITLE)) {
+			if (System.getProperty("os.name").startsWith("Windows")) {
+				splittedPath = path.split("\\\\");
+			} else {
+				splittedPath = path.split("/");
+			}
+
+			try {
+				if (splittedPath[splittedPath.length - 1].matches(EditorST.FILE_REGEX)) {
+					fileName = splittedPath[splittedPath.length - 1];
+				} else {
+					throw new ArrayIndexOutOfBoundsException("Invalid filename");
+				}
+			} catch (ArrayIndexOutOfBoundsException e) {
+				System.out.println(e.getMessage());
```

Comment:
Maybe dispatch the error message to the messageprovider if necessary at all or remove the print statement.